torch/jit/__init__.py

```diff
@@ -1250,6 +1250,9 @@ def forward(self, input):
         if not _is_new_style_class(obj):
             raise RuntimeError("TorchScript classes must be new-style classes. "
                                "Please inherit from 'object'")
+        if len(obj.mro()) > 2:
+            raise RuntimeError("TorchScript classes does not support inheritance or any polymorphism strategy yet. "
+                               "Please directly inherit from 'object'. File a issue if you want this. ")
```

Comment:
Let's keep 'File an issue' for bugs only. It implies implementing this would be easy when it is in face very difficult, which is why we haven't implemented it.